wordpress-mobile/WordPress-iOS
# Save for later: Add 'remove from saved posts' animation

Implements #9474 

An early PR to validate the general direction.
![animation_recording mov](https://user-images.githubusercontent.com/2722505/40821390-7cf1a2ca-651b-11e8-9105-606973787146.gif)

Implement removal animation and undo by not actually removing the flag marking `ReaderPost`s a saved for later, but by tracking the post that have been selected for removal, and presenting a cell containing their "collapsed state" (title, and undo button) in their place.

Caveats: 
- The design of the collapsed cell is not final. I run out of time and I wanted to validate the rest of the implementation as soon as possible.

To test:
1. Mark a few posts as saved for later
2. Navigate to Saved Posts
3. Tap the "Save for Later" button (bookmark icon) in a few posts. The posts should collapse.
4. Tap Undo. The post should be visible again
5. Navigate away from Saved Posts. Posts that were "removed" should not be visible anymore when navigating back to Saved Posts.
